lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -150,6 +178,13 @@ void addReduceOpsConversionPatterns(MLIRContext *ctx,
           typeConverter, ctx);
 }
 
+void transposeOpsConversionPatterns(MLIRContext *ctx,
```

Comment:
Transpose OP is one version of Permute, so I suggest renaming this to `addPermuteOpsConversionPatterns`

Reply:
I suggest we keep it named as transpose for now until we actually implement the permute op so the name is not misleading.

Reply:
It is currently being lowered into `ttnn.permute` and there might be other permutation ops so I thought to encompass them into one function, but I don't mind doing that later when/if we actually add those ops. Just please then rename this into `addTransposeOpsConversionPatterns` similar to other functions.

Reply:
I didn't realize there was `ttnn::permute`, since stablehlo.transpose is basically just permute, maybe it would be better to just implement `ttir.permute` and `ttnn.permute` all the way down.

Reply:
I've found this comment in `runtime/lib/ttnn/program.cpp` for TransposeOp run:
```
// Ideally this would use ttnn::transpose, but since ttnn::transpose doesn't
// work at the moment, we use this temporary solution.
...
::ttnn::Tensor out =
      ::ttnn::permute(unsqueezedInput, dimensionOrder, outputMemoryConfig);
```

Seems like they plan to support `ttnn.transpose` separately.

Reply:
We should check with someone from Metal team about this, because they already have `ttnn::transpose` OP implemented (see `third_party/tt-metal/src/tt-metal/ttnn/cpp/ttnn/operations/data_movement/transpose/transpose.hpp`). If they plan to support it long term then it is probably better to lower to `ttnn::transpose` instead of `ttnn::permute`, because it might be more optimized for transpose than generic permute OP.

Reply:
I've added comment above into the issue #652. I think you can move on with this PR as it is, lowering to `ttir::transpose`, but let's hear what @nsmithtt says.

Reply:
Yeah let's move forward with this PR and I'll follow up with my thoughts on the issue #652.